### PR TITLE
Bump to flake8 3.7.

### PR DESCRIPTION
**Pull request by anntzer**
flake8 3.7 seems to be working fine now.

The associated bump to pycodestyle 2.5 caught a few more style nits.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by dopplershift** [comment]
Is there any reason that this change should be affecting what tests we're running? Codecov.io is showing a lot of changes, and it doesn't look spurious.

**Comment by tacaswell** [comment]
It looks like appveyor did not fire, maybe this is the coverage we get only from appveyor?

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -2155,17 +2154,15 @@ def matshow(A, fignum=None, **kwargs):
 
     """
     A = np.asanyarray(A)
-    if fignum is False or fignum is 0:
+    if fignum == 0:
```
Comment (tacaswell) [inline lib/matplotlib/pyplot.py]: This is a behavior change for the case of `fignum==False`.

Comment (anntzer) [reply lib/matplotlib/pyplot.py]: No:
```
In [1]: False == 0                                                                                                                                                                                  
Out[1]: True
```

**Comment by timhoffm** [comment]
Might be that we need to take care of flake8 also in the other branches (e.g. it's failing in this backport #13340).

File: lib/matplotlib/pyplot.py (hunk 1 above)
Comment (tacaswell) [reply lib/matplotlib/pyplot.py]: fair enough.

**Comment by tacaswell** [comment]
@meeseeksbot backport to v3.0.x

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.0.x

**Comment by QuLogic** [comment]
OK, I'll put it in #13418.

**Comment by tacaswell** [comment]
Thanks @QuLogic !

**Pull request merged**